### PR TITLE
DHFPROD-5080: Reorder steps in a flow via arrows

### PR DESCRIPTION
**Pull request by devinnicholson**
### Description
This PR adds the feature of reordering steps. 
- The first step should only show a right arrow. The middle steps show both left and right arrows. The last step should only show a left arrow.
- The steps should persist after reordering.
- There was a change to updateFlowInfo api. It is now called updateFlow and contains an arg for a step array. It still updates the description the same way; that feature was not changed at all. 

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by devinnicholson** [comment]
Converting this to draft, found something I need to fix

**Review by jyangML — APPROVED** [review]

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java [hunk 1]
```diff
@@ -71,8 +72,12 @@
...
     @ApiOperation(value = "Update a flow", response = FlowInfo.class)
     @Secured("ROLE_writeFlow")
-    public ResponseEntity<JsonNode> updateFlowInfo(@PathVariable String flowName, @RequestBody UpdateFlowInfo info) {
-        return ResponseEntity.ok(newFlowService().updateFlowInfo(flowName, info.description));
+    public ResponseEntity<JsonNode> updateFlow(@PathVariable String flowName, @RequestBody JsonNode flow) {
+
+        ArrayNode stepsArray = (ArrayNode) flow.get("steps");
+        String flowDesc = flow.get("description").asText();
+
+        return ResponseEntity.ok(newFlowService().updateFlow(flowName, flowDesc, stepsArray));
```
Comment (srinathgit) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java:80]: I think you can pass the updatedFlow payload from the UI directly to the endpoint and do all the processing there. The "updatedFlow" endpoint will receive a JsonNode as input.

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/reorderFlow.sjs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+const flowService = require("../lib/flowService.sjs");
```
Comment (srinathgit) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/reorderFlow.sjs:1]: This test can might as well be called updateFlow.sjs as you have tests for updating description as well as the step order.

**Review by srinathgit — CHANGES_REQUESTED** [review]
Looks good.. One thing that needs to be added is a UI e2e test, wherein you repeatedly reorder (make sure the steps at the end are reordered correctly) the steps and finally run the steps and the output are as expected (as in in case of ingestion, explorer shows the ingested docs and in case of mapping , explorer shows the mapped docs).

**Review by wooldridge — DISMISSED** [review]

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java (hunk 1 above)
Comment (srinathgit) [reply marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/FlowController.java:80]: @devinnicholson 
Can you fix this as well ?

**Review by srinathgit — CHANGES_REQUESTED** [review]
I think you can pass the updatedFlow payload from the UI directly to the endpoint and do all the processing there. The "updatedFlow" endpoint will receive a JsonNode as input.

**Review by jyangML — APPROVED** [review]

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/updateFlow.sjs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  test.assertEqual("firstMapper-mapping", flowTest.steps["1"].stepId),
+  test.assertEqual("secondMapper-mapping", flowTest.steps["2"].stepId),
+  test.assertEqual("thirdMapper-mapping", flowTest.steps["3"].stepId)
+);
+
+// Update the description and verify
+newFlow.description = "modified";
+serviceResponse = flowService.updateFlow(newFlow);
+assertions.push(
+  test.assertEqual(newFlow.description, serviceResponse.description)
```
Comment (srinathgit) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/updateFlow.sjs:42]: Can you add a couple of asserts to make sure that step order is unchanged ?

**Review by srinathgit — CHANGES_REQUESTED** [review]
Looks good.. Just one more change needed.

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/updateFlow.sjs (hunk 3 above)
Comment (devinnicholson) [reply marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/flow/updateFlow.sjs:42]: Sure

**Review by srinathgit — APPROVED** [review]

**Review by jyangML — APPROVED** [review]

**Pull request merged**